src/Plugin/InitGraphQlTranslations.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * ScandiPWA_CmsGraphQl
+ *
+ * @category    Scandiweb
+ * @package     ScandiPWA_CmsGraphQl
+ * @copyright   Copyright (c) 2022 Scandiweb, Ltd (https://scandiweb.com)
+ */
+declare(strict_types=1);
+
+namespace ScandiPWA\CmsGraphQl\Plugin;
+
+use Exception;
+use Magento\Framework\App\Area;
+use Magento\Framework\App\FrontControllerInterface;
+use Magento\Framework\App\AreaList;
+use Magento\Framework\App\RequestInterface;
+use Magento\Framework\App\State;
+use Magento\Framework\Exception\LocalizedException;
+use ScandiPWA\PersistedQuery\Plugin\PersistedQuery;
+
+/**
+ * Class InitGraphQlTranslations
+ *
+ * A fix for https://github.com/magento/magento2/issues/31351
+ *
+ * @see \Magento\Framework\App\Action\Plugin\Design
+ * @see \Magento\Webapi\Controller\Soap
+ * @package ScandiPWA\CmsGraphQl\Plugin
+ */
+class InitGraphQlTranslations
+{
+    /**
+     * Application
+     *
+     * @var AreaList
+     */
+    protected $areaList;
```

Comment:
if this is to be merged into latest PWA, please use php 7.4 features - class property types
e.g. `protected AreaList $areaList;`